### PR TITLE
Reolink only-token client added

### PR DESCRIPTION
**Pull request by apocaliss92**

### CODE REVIEW

File: plugins/reolink/src/reolink-api.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class ReolinkCameraClient {
     parameters: Record<string, string>;
     tokenLease: number;
 
-    constructor(public host: string, public username: string, public password: string, public channelId: number, public console: Console) {
+    constructor(public host: string, public username: string, public password: string, public channelId: number, public console: Console, public forceToken?: boolean) {
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:52]: make this public readonly, forceToken can not be changed

File: plugins/reolink/src/reolink-api.ts [hunk 2]
```diff
@@ -74,13 +74,13 @@ export class ReolinkCameraClient {
     }
 
     async login() {
-        if (this.tokenLease > Date.now()) {
+        if ((!this.forceToken ? true : this.tokenLease !== Infinity) && this.tokenLease > Date.now()) {
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:77]: this change is not necessary if forceToken is readonly

File: plugins/reolink/src/reolink-api.ts [hunk 3]
```diff
@@ -89,8 +89,12 @@ export class ReolinkCameraClient {
         await this.login();
         const url = options.url as URL;
         const params = url.searchParams;
-        for (const [k, v] of Object.entries(this.parameters)) {
-            params.set(k, v);
+        if(this.forceToken) {
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:92]: this change is not necessary.

Comment (apocaliss92) [reply plugins/reolink/src/reolink-api.ts]: done, some calls fail if providing both token and user/pass, but if forceToken is set they will never be set

File: plugins/reolink/src/reolink-api.ts (hunk 2 above)
Comment (apocaliss92) [reply plugins/reolink/src/reolink-api.ts]: done

File: plugins/reolink/src/reolink-api.ts (hunk 1 above)
Comment (apocaliss92) [reply plugins/reolink/src/reolink-api.ts]: done

File: plugins/reolink/src/reolink-api.ts (hunk 3 above)
Comment (koush) [reply plugins/reolink/src/reolink-api.ts]: How would both token and username pass be set?

Comment (apocaliss92) [reply plugins/reolink/src/reolink-api.ts]: in this case it's not possible, it was something from the previous branch

**Pull request merged**